### DISCUSSION TITLE
suupport for complex mapping

### DISCUSSION THREAD

**Discussion by sangeethdba** [dbeast-co/Reindex#19] [General]
"order.return_summary.order_date": {
                            "gte": "'"$REINDEX_GTE_DATE"'",
                            "lte": "'"$REINDEX_LTE_DATE"'",
                            "format": "8uuuu-MM-dd||uuuu"

reindex api inputs:

 "source": {
                "index": "'"$SOURCE_INDEX"'",
                "type": "'"$SOURCE_INDEX_TYPE"'",
                "query": {
                "bool": {
                  "must": [
                    {
                      "range": {
                        "order.return_summary.order_date": {
                            "gte": "'"$REINDEX_GTE_DATE"'",
                            "lte": "'"$REINDEX_LTE_DATE"'",
                            "format": "8uuuu-MM-dd||uuuu"
                        }
                      }
                    }
                  ]
                }
              }
            },
            "dest": {
                "index": "'"$DESTINATION_INDEX"'",
                "type": "'"$DESTINATION_INDEX_TYPE"'",
                "op_type": "create"
            }

with mapping something like this

{
    "mappings": {
      "returns": {
        "properties": {
          "order": {
            "properties": {
              "return_summary": {
                "properties": {
                  "carrier_update_recieved_flag": {
                    "type": "keyword"
                  },
                  "customer": {
                    "properties": {
                      "city": {
                        "type": "keyword"
                      },
                      "email": {
                        "type": "keyword",
                        "normalizer": "lowercase_normalizer"
                      },
                      "name": {
                        "properties": {
                          "first": {
                            "type": "text",
                            "fields": {
                              "keyword": {
                                "type": "keyword"
                              }
                            }
                          },
                          "last": {
                            "type": "text",
                            "fields": {
                              "keyword": {
                                "type": "keyword"
                              }
                            }
                          }
                        }
                      },
                  "order_date": {
                    "type": "date"
                  },

**Comment by dbeast-co** [comment]
Hello, 

Right now we don't have the reindex_by_query support.
We plan this feature in future versions 

Best Regards
